### ISSUE TITLE
optimizations-off + testshell_markdown_typechecker = unstable

### ISSUE THREAD

**Issue by ingwinlu** [ElektraInitiative/libelektra#2057]
The recently introduced build task with optimizations disabled (#2048) seems to be unstable when it comes to testshell_markdown_typechecker.

It seems to timeout (5min of inactivity) every other run.

I have disabled ( 5ad86d4  ) the build task for now.

**Comment by markus2330** [comment]
@KurtMi any idea what is happening here?

**Comment by ingwinlu** [comment]
In the PR that added it it occured for the first time here: 
https://build.libelektra.org/jenkins/blue/organizations/jenkins/libelektra/detail/PR-2048/2/pipeline

I attributed the fail to our build servers as replaying it seemed to correct the issue in 
https://build.libelektra.org/jenkins/blue/organizations/jenkins/libelektra/detail/PR-2048/5/pipeline/339.
Interesting is that when it works it passes in 2s:
```
        Start 138: testshell_markdown_typechecker
171/221 Test #138: testshell_markdown_typechecker ...........   Passed    2.01 sec
```

**Comment by KurtMi** [comment]
> @KurtMi any idea what is happening here?

No not really

**Comment by KurtMi** [comment]
But the old jenkins job succeed up to now? or was that job deactivated?

**Comment by ingwinlu** [comment]
Old job got deactivated when we ported it over to the new system.
It never failed, but that be due to it actually not running `testshell_markdown_typechecker`:
https://build.libelektra.org/jenkins/job/elektra-gcc-configure-debian-optimizations/299/consoleFull

**Comment by KurtMi** [comment]
Alright, but in the PR #2055 it worked the first time:
https://build.libelektra.org/jenkins/blue/organizations/jenkins/libelektra/detail/PR-2055/10/pipeline/

**Comment by ingwinlu** [comment]
@KurtMi you think the changes in #2055 might contain a fix? because I can not spot anything that would affect the behaviour. I requested another build of #2055 to see if we can make it fail identically.

**Comment by KurtMi** [comment]
> @KurtMi you think the changes in #2055 might contain a fix?

No, but it failed always the first time (or not?) and in #2055 not.

**Comment by markus2330** [comment]
Are we sure that the problems are caused by turning off optimizations in the build?

If this is not the case, @KurtMi is not the right person :wink:

**Comment by KurtMi** [comment]
Well we might ask @e1528532 if he has any clue. I tried for the last hour to compile the typechecker plugin locally but I failed :)

**Comment by e1528532** [comment]
It might be related to the issue adressed in #2055 i try to get that merged today

**Comment by KurtMi** [comment]
> @KurtMi have you installed c2hs today or recently in order to compile it?

Yes, i followed the instructions from `/src/bindings/haskell/README.md`

**Comment by e1528532** [comment]
> Yes, i followed the instructions from /src/bindings/haskell/README.md

I think thats the issue. A new version of language-c, which is a dependency of c2hs got releast on the 2nd june. Unfortunately if c2hs gets compiled with that version, it fails to build the typechecker plugin. With the previous language-c version 0.7.2 it seems to work.

So if you want to try it out locally, first ensure the previous version is removed with the command
`ghc-pkg unregister language-c`
and then install the previous version of language-c with the command `cabal install language-c-0.7.2' and last reinstall c2hs with 'cabal install c2hs'. In that case it should compile with that version. Now if you cleanup the previous build artifacts it should be able to compile it again.

Good point thought i have to add that to the readme as well!

**Comment by KurtMi** [comment]
@ingwinlu it seems that the retrigger of PR-2055 also worked.
https://build.libelektra.org/jenkins/blue/organizations/jenkins/libelektra/detail/PR-2055/11/pipeline
